examples/seq2seq/train.py

```diff
@@ -56,21 +56,39 @@
 class CharacterTable(object):
   """Encode/decodes between strings and integer representations."""
 
+  @property
+  def pad_id(self):
+    return 0
+
+  @property
+  def eos_id(self):
+    return 1
+
+  @property
+  def vocab_size(self):
+    return len(self._chars) + 2
+
   def __init__(self, chars):
     self._chars = sorted(set(chars))
-    self._char_indices = dict((ch, idx) for idx, ch in enumerate(self._chars))
-    self._indices_char = dict((idx, ch) for idx, ch in enumerate(self._chars))
+    self._char_indices = dict(
+        (ch, idx + 2) for idx, ch in enumerate(self._chars))
+    self._indices_char = dict(
+        (idx + 2, ch) for idx, ch in enumerate(self._chars))
+    self._indices_char[self.pad_id] = '_'
```

Comment:
`?` is more commonly used for unk, but we could use it here if you'd prefer. We could also do `<pad>`, which is common. Let me know your preference.

Reply:
I don't have a strong preference, I guess I was thinking of ' ' since I used that when padding (because I was printing it), but given that we won't use it anyway, '_' is fine to me.